synedrion/src/uint/secret_signed.rs

```diff
@@ -0,0 +1,710 @@
+    }
+
+    pub fn bound(&self) -> u32 {
+        self.bound
+    }
+
+    pub fn to_public(&self) -> PublicSigned<T> {
+        PublicSigned::new_from_unsigned(self.value.expose_secret().clone(), self.bound).expect("the bound is valid")
+    }
+
+    /// Performs the unary `-` operation.
+    pub fn neg(&self) -> Self {
+        Self {
+            value: self.value.wrapping_neg(),
+            bound: self.bound,
+        }
+    }
+}
+
+impl<T> SecretSigned<T>
+where
+    T: ConditionallySelectable + Zeroize + Integer + Bounded,
+{
+    pub fn conditional_select(a: &Self, b: &Self, choice: Choice) -> Self {
+        Self {
+            bound: u32::conditional_select(&a.bound, &b.bound, choice),
+            value: Secret::<T>::conditional_select(&a.value, &b.value, choice),
+        }
+    }
+
+    pub fn abs_value(&self) -> Secret<T> {
+        Secret::<T>::conditional_select(&self.value, &self.value.wrapping_neg(), self.is_negative())
+    }
+
+    /// Computes the absolute value of [`self`]
+    pub fn abs(&self) -> SecretUnsigned<T> {
+        SecretUnsigned::new(
+            Secret::<T>::conditional_select(&self.value, &self.value.wrapping_neg(), self.is_negative()),
+            self.bound,
+        )
+        .expect("the absolute value is within the same bound")
+    }
+
+    /// Creates a [`SignedSecret`] from an unsigned value, treating it as if it encodes the sign as two's complement.
+    ///
+    /// Returns `None` if the requested bound is too large, or if `abs(value)` is actually larger than the bound.
+    pub fn new_from_unsigned(value: Secret<T>, bound: u32) -> Option<Self> {
+        let is_negative = Choice::from(value.expose_secret().bit_vartime(T::BITS - 1) as u8);
+        let abs = Secret::<T>::conditional_select(&value, &value.wrapping_neg(), is_negative);
+        if bound >= T::BITS || abs.expose_secret().bits() > bound {
+            return None;
+        }
+        Some(Self::new_from_unsigned_unchecked(value, bound))
+    }
+
+    /// Creates a [`SignedSecret`] from an unsigned value, treating it as if it is the absolute value.
+    /// If `is_negative` is truthy, crates a negative [`SignedSecret`] with the given absolute value.
+    ///
+    /// Returns `None` if the bound is too large, or if `abs_value` is actually larger than the bound.
+    fn new_from_abs(abs_value: Secret<T>, bound: u32, is_negative: Choice) -> Option<Self> {
+        Self::new_from_unsigned(
+            Secret::<T>::conditional_select(&abs_value, &abs_value.wrapping_neg(), is_negative),
+            bound,
+        )
+    }
+
+    #[cfg(test)]
+    fn test_new_from_abs(abs_value: T, bound: u32, is_negative: bool) -> Option<Self> {
+        Self::new_from_abs(Secret::init_with(|| abs_value), bound, Choice::from(is_negative as u8))
+    }
+
+    #[cfg(test)]
+    fn test_new_from_unsigned(abs_value: T, bound: u32) -> Option<Self> {
+        Self::new_from_unsigned(Secret::init_with(|| abs_value), bound)
+    }
```

Comment:
Could these be moved to the `tests` module you think?

Reply:
Yes, I suppose